relayer/relays/fisherman/beefy-listener.go

```diff
@@ -0,0 +1,488 @@
+						// combine payload1 and payload2
+						payload := []interface{}{types.NewData(append(payload1, payload2...))}
+						c, err := types.NewCall(meta, extrinsicName, payload...)
+						// c, err := types.NewCall(meta, extrinsicName, types.NewBytes(payload1), types.NewBytes(payload2))
+						if err != nil {
+							return fmt.Errorf("create call: %w", err)
+						}
+
+						ext, err := li.signedExtrinsicFromCall(meta, c)
+
+						// Send the extrinsic
+						sub, err := li.relaychainConn.API().RPC.Author.SubmitAndWatchExtrinsic(ext)
+						// res, err := li.relaychainConn.API().RPC.Author.SubmitExtrinsic(ext)
+						if err != nil {
+							log.Error("Failed to submit extrinsic: ", err, sub)
+						} else {
+							err := li.watchExtrinsicSubscription(sub)
+							log.Info("Extrinsic submitted: ", sub)
+							if err != nil {
+								log.Error("Extrinsic submission failed: ", err)
+							} else {
+								log.Info("Equivocation report complete")
+							}
+						}
+
+					} else {
+						canonicalBlockHash, err := li.relaychainConn.API().RPC.Chain.GetBlockHash(event.BlockNumber)
+						if err != nil {
+							return fmt.Errorf("fetch block hash: %w", err)
+						}
+						canonicalMmrRootHash, err := li.relaychainConn.GetMMRRootHash(canonicalBlockHash)
+						if err != nil {
+							return fmt.Errorf("retrieve MMR root hash at block %v: %w", canonicalBlockHash.Hex(), err)
+						} else {
+							validatorProof, err := li.parseSubmitInitialProof(callData)
+							if err != nil {
+								log.WithError(err).Warning("Failed to decode transaction call data")
+							}
+							log.WithFields(log.Fields{
+								"commitment":               commitment,
+								"commitment.payload.data":  fmt.Sprintf("%#x", commitment.Payload[0].Data),
+								"correspondingMMRRootHash": canonicalMmrRootHash,
+							}).Debug("Decoded transaction call data for NewTicket event")
+							if canonicalMmrRootHash != types.NewHash(commitment.Payload[0].Data) {
+								log.WithFields(log.Fields{
+									"commitment.payload.data":  fmt.Sprintf("%#x", commitment.Payload[0].Data),
+									"proof":                    validatorProof,
+									"correspondingMMRRootHash": canonicalMmrRootHash,
+								}).Warning("MMR root hash does NOT match the commitment payload")
+
+								meta, err := li.relaychainConn.API().RPC.State.GetMetadataLatest()
+								if err != nil {
+									return fmt.Errorf("get metadata: %w", err)
+								}
+
+								extrinsicName := "Beefy.report_fork_voting"
+								// call: c803
+								// build vote payload for equivocation proof
+								offenderPubKeyCompressed, offenderSig, err := getOffenderPubKeyAndSig(commitment, validatorProof)
+								if err != nil {
+									return fmt.Errorf("get offender pubkey and sig: %w", err)
+								}
+
+								payload1 := constructVotePayload(commitment, offenderPubKeyCompressed, offenderSig)
+
+								// Ancestry Proof
+								payload2, err := li.constructAncestryProofPayload(commitment)
```

Comment:
Is `payload2` supposed to be generated against the current finalized Beefy block? i.e.,

```
payload2, err := li.constructAncestryProofPayload(commitment,latestHash)
```